Project/searchResult.js

```diff
@@ -0,0 +1,310 @@
+        this.searchContainer.prepend(this.comparisonDiv)
+        this.compareXCompanies = document.createElement('div')
+        this.compareXCompanies.id = 'comparisonX'
+        this.searchContainer.prepend(this.compareXCompanies)
+        this.compareCount;
+        this.allButtons = document.querySelectorAll('#comparisonButton')
+
+
+    }
+
+    // Function to add and remove the compare company count
+    addAndRemoveList(contentToAdd) {
+
+        let companiesCountCheckMaxThree = document.querySelectorAll('.comparisonButton')
+        // Max 3 companies to compare limit
+        if (companiesCountCheckMaxThree.length > 2) {
+            return alert('You can compare max 3 companies')
+        } else {
+            // Check for duplicate companies before adding new button and then alert
+            let duplicateFound = false;
+            for (let i = 0; i < companiesCountCheckMaxThree.length; i++) {
+                if (companiesCountCheckMaxThree[i].innerText.includes(contentToAdd)) {
+                    duplicateFound = true;
+                    break;
+                }
+            }
+            if (duplicateFound) {
+                return alert(`You've already added ${contentToAdd} to the comparison list.`);
+            }
+        }
+
+
+        this.comparisonDiv.innerHTML += `<button class="comparisonButton">${contentToAdd} x </button>`
+
+        let companiesCount = document.querySelectorAll('.comparisonButton')
+        this.compareCount = companiesCount.length
+
+        let accumulatedSymbols = []
+
+        for (let i = 0; i < companiesCount.length; i++) {
+
+            accumulatedSymbols.push(companiesCount[i].innerText.split(" ", 1).join(""))
+            companiesCount[i].addEventListener('click', () => {
+                companiesCount[i].remove()
+                this.compareCount--
+                this.updateCompareUrl()
+            })
+        }
+
+        this.updateCompareUrl()
+
+    }
+
+    // To add the a tag that will redirect the user to the compare page
+    updateCompareUrl() {
+        let companiesCount = document.querySelectorAll('.comparisonButton')
+        let accumulatedSymbols = []
+
+        for (let i = 0; i < companiesCount.length; i++) {
+            accumulatedSymbols.push(companiesCount[i].innerText.split(" ", 1))
+        }
+
+
+        const UrlString = accumulatedSymbols.join(",")
+
+
+        this.compareXCompanies.innerHTML = `<a href="compare.html?symbols=${UrlString}">Compare ${companiesCount.length} Companies</a>`
+    }
+
+
+}
+
+
+class SearchResults extends CompareStocks {
+
+    constructor(element) {
+
+        super()
+        // Declare all the variables to be worked with
+
+        this.element = element
+        // Creating all the elements needed for html
+        let listContainer = document.createElement('ul')
+        let searchButton = document.createElement('button')
+        let searchContainer = document.getElementById('searchContainer')
```

Comment:
if the searchContainer of any instance of this class will be the same container, then it loses the purpse and abilities of a class, because in real situation you can only create one instace without facing a real probelm.
The solution - pass the container as a parameter to the constractor function.